### ISSUE TITLE
NTP binding: hostname parameter not saved

### ISSUE THREAD

**Issue by lolodomo** [eclipse/smarthome#864]
Using PaperUI (openHAB2 beta 1), I installed the NTP binding. A ntp:ntp:local thing was created. Then I edited its parameters to switch from default server 0.pool.ntp.org to 0.fr.pool.ntp.org. If I open again the parameters, the value is 0.fr.pool.ntp.org as expected.
Now if I stop OH2 (logout command in the console) and start again OH2, when I edit my NTP thing parameters, my hostname is now 0.pool.ntp.org rather than 0.fr.pool.ntp.org

**Comment by marcelrv** [comment]
@lolodomo thanks for your finding.
This seems some sort of issue in the persistence of config parameters.
I'm not sure if this is directly related to the binding itself.

I'll try to replicate this here on my instance.

**Comment by kaikreuzer** [comment]
I can reproduce it and @marcelrv is right - this has nothing to do with the binding, but seems to be a bug in the core framework. I will analyze it further...

**Comment by marcelrv** [comment]
I can replicate this issue, in fact it it applies to any config parameter, not only hostname.
**I can confirm this not only applies to NTP binding but applies to other bindings as well**, e.g. I have replicated this issue in the yahoo binding. 
NB, it is not always, so I guess there is some timing issue

* during the session parameters are correctlyu updated, binding is using updated values.
* after shutdown and restart the values are lost. binding fall back to the default values from xml in ESH-INF

**Comment by kaikreuzer** [comment]
for me, I do not even have to restart to see the old values...

**Comment by kaikreuzer** [comment]
I found the problem, it is after all in the NTP binding:
If an existing thing is re-discovered, updated configuration from the discovery result is automatically sync'ed to the thing. As the NTP discovery set the default values, this overwrote any changes upon a rediscovery (which was automatically triggered at system startup).
I have removed this now in the NtpDiscovery in #894

**Comment by marcelrv** [comment]
@kaikreuzer maybe we need to move it to the forum, but I would think that is conceptional not the right approach. This way somehow the discovery should be made aware of all the changes.
I would assume that discovery does what it says: discovery. From that point onwards it is up to the user to update the configuration in either the binding or via the UI. It would be very hairy if we mix these.

e.g. take here the locale, there is no default set in the esh xml file, nor should there hence with the change made it will be left blank, 
I intentionally made it in the way it is as to have the clear example of how this field should be populated (formatted).

I think there will be many other examples as well where values set during discovery will be updated by the user later onwards. (same thing for e.g. the yahoo binding, I updated the location ID, to the one better reflecting my location, but it got changed back by discovery...not what I want)
